corehq/apps/change_feed/partitioners.py

```diff
@@ -66,3 +68,11 @@ def _get_topic_to_pillow_map():
         for topic, pillow_name in TOPIC_TO_PILLOW_NAME_MAP.items()
         if pillow_name in all_pillow_names
     }
+
+
+@memoized
+def get_kafka_consumer_for_partitioning():
+    return ClosingContextProxy(KafkaConsumer(
```

Comment:
`ClosingContextProxy` is not a re-entrant context manager, so I thing this will not work with `@memoized`.

https://github.com/dimagi/commcare-hq/blob/6b64ab191f8307f5e9977e785232a73aa27491c0/corehq/util/io.py#L18-L22

Reply:
I see, I misunderstood how this works. Thanks for pointing that out.

Can you see an issue with memoizing it, not using `ClosingContextProxy` + `with consumer:`, and just using a single kafka consumer that we never close? I imagine it closes when the OS process ends anyway, so the point of closing is for when we're creating a large / unbounded number of different connections over the life of the OS process? But maybe I'm wrong about that

Reply:
I don't know how Kafka connections work, but that would not be a good pattern for a PostgreSQL connection. For example, if the connection ever got broken then it would likely be permanently broken until the process is restarted.

Reply:
It's not clear if you're intending to do more here or if your most recent comment was a suggestion that you'd like to keep the code as it is. I am fairly certain the code will not work as-is. Either `@memoized` or `ClosingContextProxy` should be removed.

Reply:
Sorry yeah I was intending to do more on it. I agree with you it won't work as is